### ISSUE TITLE
Pyright gives an error when using a `Literal` typehint for the result of `getattr` with a default

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2548]
**Describe the bug**
Pyright gives an error when using a `Literal` typehint for the result of `getattr`, even though the default matches the `Literal` typehint. Mypy does not give an error for this. This is not the case when the literal typehint only has one type argument that matches the default, e.g. `Literal[3]` for the example code.

**Expected behavior**
The following code to successfully type check.

**Screenshots or Code**
```python
from typing import Literal

x: Literal[1, 2, 3] = getattr(object(), '', 3)
```
```
error: Argument of type "Literal[3]" cannot be assigned to parameter "__default" of type "_T@getattr" in function "getattr"
    Type "Literal[1]" cannot be assigned to type "Literal[3]" (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (pyright 1.1.185)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

Pyright includes some heuristics for dealing with type compatibility checks that involve unions with type variables. In this particular case, the return type of `getattr` is a union of `Any` and `T`, and one of the heuristics was failing. The result was that only `Literal[1]` was assigned to `T` and `Literal[2, 3]` was assigned to `Any`.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.186, which I just published. It will also be included in the next release of pylance.
